File: stress/t/arr_matmul8.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t n,r,c,k,acc,a,b,t,s;
  int64_t A[64],B[64],C[64];
  n=8;
  for(r=0;r<n;r++){
    for(c=0;c<n;c++){
      A[r*n+c]=(r+1)*(c+2)%13;
      B[r*n+c]=(r*3+c+1)%11;
    }
  }
  for(r=0;r<n;r++){
    for(c=0;c<n;c++){
      acc=0;
      for(k=0;k<n;k++){ a=A[r*n+k]; b=B[k*n+c]; acc=acc+a*b; }
      C[r*n+c]=acc;
    }
  }
  s=0;
  for(r=0;r<n;r++){ t=C[r*n+r]; s=s+t; }
  printf("diag="); printf("%ld",(long)s); printf("\n");
  t=C[0]; printf("c00="); printf("%ld",(long)t); printf("\n");
  t=C[63]; printf("c77="); printf("%ld",(long)t); printf("\n");
  return 0;
}

File: stress/t/arr_matmul8.f
      PROGRAM ARRMATMUL8
      INTEGER*8 N,R,C,K,ACC,A,B,T,S
      INTEGER*8 AR(0:63),BR(0:63),CR(0:63)
      N=8
      R=0
   10 IF (R.GE.N) GOTO 30
      C=0
   20 IF (C.GE.N) GOTO 25
      AR(R*N+C)=MOD((R+1)*(C+2),13_8)
      BR(R*N+C)=MOD(R*3+C+1,11_8)
      C=C+1
      GOTO 20
   25 CONTINUE
      R=R+1
      GOTO 10
   30 CONTINUE
      R=0
   40 IF (R.GE.N) GOTO 70
      C=0
   50 IF (C.GE.N) GOTO 65
      ACC=0
      K=0
   55 IF (K.GE.N) GOTO 60
      A=AR(R*N+K)
      B=BR(K*N+C)
      ACC=ACC+A*B
      K=K+1
      GOTO 55
   60 CONTINUE
      CR(R*N+C)=ACC
      C=C+1
      GOTO 50
   65 CONTINUE
      R=R+1
      GOTO 40
   70 CONTINUE
      S=0
      R=0
   80 IF (R.GE.N) GOTO 90
      T=CR(R*N+R)
      S=S+T
      R=R+1
      GOTO 80
   90 CONTINUE
      WRITE(*,'(A,I0)') 'diag=',S
      T=CR(0)
      WRITE(*,'(A,I0)') 'c00=',T
      T=CR(63)
      WRITE(*,'(A,I0)') 'c77=',T
      END
